data/test/scenarios/lua_tests/wesnoth/test_wfl_rotate_loc_around.cfg

```diff
@@ -0,0 +1,12 @@
+{GENERIC_UNIT_TEST "test_wfl_rotate_loc_around" (
+[event]
+      name = start
+      [lua]
+            code=<<
+wml.variables.test_ok = wesnoth.eval_formula(wesnoth.compile_formula("loc(9,10)")).y == wesnoth.eval_formula(wesnoth.compile_formula("rotate_loc_around(loc(10,10),loc(9,11),1)")).y and wesnoth.eval_formula(wesnoth.compile_formula("loc(9,10)")).x == wesnoth.eval_formula(wesnoth.compile_formula("rotate_loc_around(loc(10,10),loc(9,11),1)")).x
```

Comment:
looks correct, but I would simplify wesnoth.eval_formula(wesnoth.compile_formula("loc(9,10)")).y to 10 and wesnoth.eval_formula(wesnoth.compile_formula("loc(9,10)")).x to 9

Reply:
and wesnoth.eval_formula(wesnoth.compile_formula("rotate_loc_around(loc(10,10),loc(9,11),1)")) can be in variable to not need to evaluate it twice

Reply:
I think this is a really bad way of writing it.

1. Please use the `unit_test` module, specifically `unit_test.assert_equal`. You don't need to join them into a single condition with `and`, because multiple assertions all must pass anyway.
2. The condition itself is unnecessarily verbose for several reasons, which I'll explain in more detail below.
3. This approach also means you don't need a variable or a `FAIL_IF_NOT`, as the assertion will terminate the test if it fails.

On to an explanation of what you're doing in the condition that is sub-optimal.

Although `wesnoth.eval_formula` can indeed take a compiled formula as you've done here, I think there's rarely a reason to actually do that. It's more useful for a one-time formula, because if you pass it a string, it automatically compiles it _and_ evaluates it. On the other hand, if you have a compiled formula in a variable, you can call it just like a function.

So, what I'd recommend doing here is to compile each of the two formulas once and save them to a Lua variable. Then, write your assertions using those functions just like a normal function.

Reply:
I encouraged ZombieKnight to create test to find out if the change helps, since it would require being able to compile game otherwise. If the PR would be better without test, we can just change index and not add test.